src/libsyncengine/jobs/network/API_v2/downloadjob.cpp

```diff
@@ -493,95 +491,79 @@ bool DownloadJob::createLink(const std::string &mimeType, const std::string &dat
     return true;
 }
 
-bool DownloadJob::removeTmpFile(const SyncPath &path) {
-    if (std::error_code ec; !std::filesystem::remove_all(path, ec)) {
-        LOGW_WARN(_logger, L"Failed to remove all: " << Utility::formatStdError(path, ec));
+bool DownloadJob::removeTmpFile() {
+    if (_tmpPath.empty()) return true;
+
+    if (std::error_code ec; !std::filesystem::remove_all(_tmpPath, ec)) {
+        LOGW_WARN(_logger, L"Failed to remove a downloaded temporary file: " << Utility::formatStdError(_tmpPath, ec));
         return false;
     }
 
     return true;
 }
 
-bool DownloadJob::moveTmpFile(const SyncPath &tmpPath, bool &restartSync) {
+bool DownloadJob::moveTmpFile(bool &restartSync) {
     restartSync = false;
 
     // Move downloaded file from tmp directory to sync directory
-    std::error_code ec;
 #ifdef _WIN32
     bool retry = true;
     int counter = 50;
     while (retry) {
         retry = false;
 #endif
 
-        IoError ioError = IoError::Success;
-        IoHelper::moveItem(tmpPath, _localpath, ioError);
-        const bool crossDeviceLink = ioError == IoError::CrossDeviceLink;
-        if (ioError != IoError::Success && !crossDeviceLink) {
-            LOGW_WARN(_logger, L"Failed to move: " << Utility::formatIoError(tmpPath, ioError) << L" to "
-                                                   << Utility::formatSyncPath(_localpath));
-            return false;
-        }
+        // Copy file content (i.e. doesn't change node id)
+        std::error_code ec;
+        std::filesystem::copy(_tmpPath, _localpath, std::filesystem::copy_options::overwrite_existing, ec);
+        if (ec) {
+            LOGW_WARN(_logger,
+                      L"Failed to copy downloaded file to " << Utility::formatSyncPath(_localpath) << L", " << Utility::formatStdError(ec));
 
-        if (crossDeviceLink) {
```

Comment:
Is `std::filesystem::copy` significantly slower than `std::filesystem::rename`? If so, perhaps we could keep it, as https://github.com/Infomaniak/desktop-kDrive/pull/442 resolves the issue of ID reuse. Otherwise, removing it would indeed enhance security.

Reply:
I suppose that rename is better than copy if the target doesn't exist (CREATE case) and the source and target paths are on the same partition.
If the target exists (EDIT case), I don't know... In any case, I would prefer copy as the node id doesn't change (fewer modifications to the DB, update tree etc.).

Reply:
Updated to use the rename for a CREATE operation (source and target on the same FS).
The copy is used for a CREATE operation (source and target on distinct FSs) or an EDIT operation.